scipy/stats/tests/test_distributions.py

```diff
@@ -2302,33 +2302,26 @@ def test_reduces_to_triang(self):
 
     def test_reduces_to_uniform(self):
         x = np.linspace(0, 1, 10)
-        with suppress_warnings() as sup, np.errstate(divide='ignore'):
-            sup.filter(RuntimeWarning,
-                       "invalid value encountered in true_divide")
-            assert_almost_equal(stats.trapz.pdf(x, 0, 1), stats.uniform.pdf(x))
-            assert_almost_equal(stats.trapz.cdf(x, 0, 1), stats.uniform.cdf(x))
+        assert_almost_equal(stats.trapz.pdf(x, 0, 1), stats.uniform.pdf(x))
+        assert_almost_equal(stats.trapz.cdf(x, 0, 1), stats.uniform.cdf(x))
 
     def test_cases(self):
-        with suppress_warnings() as sup, np.errstate(divide='ignore'):
-            sup.filter(RuntimeWarning,
-                       "invalid value encountered in true_divide")
-
-            # edge cases
-            assert_almost_equal(stats.trapz.pdf(0, 0, 0), 2)
-            assert_almost_equal(stats.trapz.pdf(1, 1, 1), 2)
-            assert_almost_equal(stats.trapz.pdf(0.5, 0, 0.8), 1.11111111111111111)
-            assert_almost_equal(stats.trapz.pdf(0.5, 0.2, 1.0), 1.11111111111111111)
-
-            # straightforward case
-            assert_almost_equal(stats.trapz.pdf(0.1, 0.2, 0.8), 0.625)
-            assert_almost_equal(stats.trapz.pdf(0.5, 0.2, 0.8), 1.25)
-            assert_almost_equal(stats.trapz.pdf(0.9, 0.2, 0.8), 0.625)
-
-            assert_almost_equal(stats.trapz.cdf(0.1, 0.2, 0.8), 0.03125)
-            assert_almost_equal(stats.trapz.cdf(0.2, 0.2, 0.8), 0.125)
-            assert_almost_equal(stats.trapz.cdf(0.5, 0.2, 0.8), 0.5)
-            assert_almost_equal(stats.trapz.cdf(0.9, 0.2, 0.8), 0.96875)
-            assert_almost_equal(stats.trapz.cdf(1.0, 0.2, 0.8), 1.0)
+        # edge cases
+        assert_almost_equal(stats.trapz.pdf(0, 0, 0), 2)
+        assert_almost_equal(stats.trapz.pdf(1, 1, 1), 2)
```

Comment:
Hi,

I am wondering why the cases c=d=0 and c=d=1 are allowed. In that case, the pdf is not a density (does not integrate to one) / the distribution is discrete while it is implmented as a continuous Distribution. Wouldn't it be better to exclude this case (_argcheck)?

And one could add a test case for c=d in ]0,1[ as a special case.

Reply:
The cases you describe are specific examples of `c == d`. When `c == d` the trapezoidal distribution reduces to the `stats.triang` continuous distribution, with the mode at `c`. For example:

```
from scipy.stats import trapz
import matplotlib.pyplot as plt
x=np.linspace(0.000001, 0.99999999, 10000)
plt.plot(x, trapz.pdf(x, 0.001, 0.001))
from scipy.integrate import quadrature
print(quadrature(trapz.pdf, 0, 1, (0.00000001, 0.00000001), tol=1e-15, rtol=1e-15))
```

For the case of `c=d=0` then the mode should be at 0, with `trapz.pdf(0, 0, 0) = 2`
The integral should integrate to 1 - the triangles base is 1 in length, 2 in height. Area = 2 * 1 / 2 = 1. I'm not sure why you think it's a discrete distribution.

Reply:
Right, I had something different in mind but c=d=0 and c=d=1 work fine like you said.
Would it make sense to add a test case of the special case c=d to check that trapz and triang are indeed the same?

Reply:
It does make sense to have those tests, they're already there as `test_reduces_to_triang`